app/src/main/res/layout/activity_main.xml

```diff
@@ -4,15 +4,101 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:visibility="visible"
     tools:context=".MainActivity">
 
     <TextView
+        android:id="@+id/city_name"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
+        android:layout_marginTop="12dp"
+        android:text="@string/st_paul_mn"
+        android:textSize="18sp"
+        android:textColor="@color/black"
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
+    <TextView
+        android:id="@+id/temperature"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textSize="72sp"
+        android:layout_marginTop="12dp"
+        android:layout_marginStart="48dp"
+        android:textColor="@color/black"
+        app:layout_constraintTop_toBottomOf="@id/city_name"
+        app:layout_constraintStart_toStartOf="parent"
+        tools:text="@string/temperature" />
```

Comment:
`tools:text` does not set the value of the TextView at runtime. It is only for design time use so the design preview has a display value.